connect/api/src/main/java/org/apache/kafka/connect/health/TaskState.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.connect.health;
+
+import java.util.Objects;
+
+/**
+ * {@link TaskState} provides the state, ids and any errors associated with Connector
+ * tasks.
+ */
+public class TaskState extends AbstractState implements Comparable<TaskState> {
+
+    private final int taskId;
+
+    public TaskState(int taskId, String state, String workerId, String msg) {
```

Comment:
Add JavaDoc, since this is part of the public API. Which of the parameters are allowed to be null and/or empty?